src/Microsoft.ML.Data/Data/RowCursorUtils.cs

```diff
@@ -341,7 +341,7 @@ public static string TestGetLabelGetter(DataViewType type, bool allowKeys)
             if (allowKeys && type is KeyType)
                 return null;
 
-            return allowKeys ? "Expected R4, R8, Bool or Key type" : "Expected R4, R8 or Bool type";
+            return allowKeys ? "Expected Single, Double, Boolean or Key type" : "Expected Single, Double or Boolean type";
```

Comment:
>Single [](start = 90, length = 6)

 i feel like "float" "double" or "bool" are even more user friendly.  

we don't usually go :
Boolean k = true;

Reply:
from C# developer point, yes.
But as soon as you work with TextLoader or you inside Visual Basic you no longer have `bool` our Datakind is `Boolean` CLS wise i think it's also `Boolean`

---
In reply to: [268223385](https://github.com/dotnet/machinelearning/pull/3046#discussion_r268223385) [](ancestors = 268223385)

Reply:
For our C# code, we prefer to use the language specific keywords, like `float`, `bool`, `short`, etc.

However, for public APIs, strings, etc. where these words/names are visible to other languages we shouldn't be using the C# language specific keywords. Instead we should be using the .NET Type names like `Boolean`, `Single`, `Int16`, etc. We made those changes in our public API, like the `public DataKind enum`. We should use the same names in these strings.